cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,241 @@
+	// ApplicationOrder is the index of the transaction among all the transactions
+	// for that ledger.
+	ApplicationOrder int32 `json:"applicationOrder"`
+	// FeeBump indicates whether the transaction is a feebump transaction
+	FeeBump bool `json:"feeBump"`
+	// EnvelopeXdr is the TransactionEnvelope XDR value.
+	EnvelopeXdr string `json:"envelopeXdr"`
+	// ResultXdr is the TransactionResult XDR value.
+	ResultXdr string `json:"resultXdr"`
+	// ResultMetaXdr is the TransactionMeta XDR value.
+	ResultMetaXdr string `json:"resultMetaXdr"`
+	// DiagnosticEventsXDR is present only if transaction was not successful.
+	// DiagnosticEventsXDR is a base64-encoded slice of xdr.DiagnosticEvent
+	DiagnosticEventsXDR []string `json:"diagnosticEventsXdr,omitempty"`
+	// Ledger is the sequence of the ledger which included the transaction.
+	Ledger uint32 `json:"ledger"`
+	// LedgerCloseTime is the unix timestamp of when the transaction was included in the ledger.
+	LedgerCloseTime int64 `json:"createdAt"`
+}
+
+// GetTransactionsResponse encapsulates the response structure for getTransactions queries.
+type GetTransactionsResponse struct {
+	Transactions          []TransactionInfo `json:"transactions"`
+	LatestLedger          uint32            `json:"latestLedger"`
+	LatestLedgerCloseTime int64             `json:"latestLedgerCloseTimestamp"`
+	OldestLedger          uint32            `json:"oldestLedger"`
+	OldestLedgerCloseTime int64             `json:"oldestLedgerCloseTimestamp"`
+	Cursor                string            `json:"cursor"`
+}
+
+type transactionsRPCHandler struct {
+	ledgerReader      db.LedgerReader
+	dbReader          db.TransactionReader
+	maxLimit          uint
+	defaultLimit      uint
+	logger            *log.Entry
+	networkPassphrase string
+}
+
+// getTransactionsByLedgerSequence fetches transactions between the start and end ledgers, inclusive of both.
+// The number of ledgers returned can be tuned using the pagination options - cursor and limit.
+func (h transactionsRPCHandler) getTransactionsByLedgerSequence(ctx context.Context, request GetTransactionsRequest) (GetTransactionsResponse, error) {
+	ledgerRange, err := h.dbReader.GetLedgerRange(ctx)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InternalError,
+			Message: err.Error(),
+		}
+	}
+
+	err = request.isValid(h.maxLimit, ledgerRange)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidRequest,
+			Message: err.Error(),
+		}
+	}
+
+	// Move start to pagination cursor
+	start := toid.New(int32(request.StartLedger), 1, 1)
+	limit := h.defaultLimit
+	if request.Pagination != nil {
+		if request.Pagination.Cursor != "" {
+			cursorInt, err := strconv.ParseInt(request.Pagination.Cursor, 10, 64)
+			if err != nil {
+				return GetTransactionsResponse{}, &jrpc2.Error{
+					Code:    jrpc2.InvalidParams,
+					Message: err.Error(),
+				}
+			}
+
+			*start = toid.Parse(cursorInt)
```

Comment:
I think you should validate the cursor here

Reply:
@tamirms Is this what you had in mind for validation?

`if start.String() != request.Pagination.Cursor {
		return GetTransactionsResponse{}, &jrpc2.Error{
					Code:    jrpc2.InternalError,
					Message: fmt.Sprintf("could not parse cursor: expected %s, got %s", request.Pagination.Cursor, start.String()),
				}
}`

Reply:
@aditya1702 sorry, I think you can ignore this comment. I think the check above `cursorInt, err := strconv.ParseInt(request.Pagination.Cursor, 10, 64)` should cover it